src/main/scala/com/fulcrumgenomics/vcf/HapCutToVcf.scala

```diff
@@ -73,19 +72,30 @@ import scala.collection.JavaConverters._
       |  3. The "MCL" tag gives the maximum change in likelihood if this SNP is made homozygous or removed.
       |  4. The "RMEC" tag gives the reduction in MEC score if we remove this variant altogether.
       |
+      |Additional FORMAT fields for phased variants are provided corresponding to per-genotype information produced by
+      |HAPCUT2:
+      |  1. The "PR" tag is 1 if HapCut2 pruned this variant, 0 otherwise.
+      |  2. The "SE" tag gives the confidence (log10) that there is not a switch error occurring immediately before the SNV
+      |  3. The "NE" tag gives the confidence (log10) that the SNV is not a mismatch (single SNV) error.
+      |HapCut2 should not be run with `--call_homozygous` as the genotypes may be different than the input and as not
+      |currently supported.
+      |
       |For more information about HAPCUT, see the source code or paper below.
       |  source code: https://github.com/vibansal/hapcut
       |  HapCut paper: An efficient and accurate algorithm for the haplotype assembly problem Bioinformatics. 2008 Aug
       |    15;24(16):i153-9.
       |
+      |For more information about HAPCUT2, see the source code below.
+      |   source code: https://github.com/pjedge/hapcut2
     """,
   group=ClpGroups.VcfOrBcf
 )
 class HapCutToVcf
-( @arg(flag="v", doc="The original VCF provided to HAPCUT.") val vcf: PathToVcf,
-  @arg(flag="i", doc="The output file from HAPCUT.") val input: FilePath,
+( @arg(flag="v", doc="The original VCF provided to HAPCUT/HAPCUT2.") val vcf: PathToVcf,
+  @arg(flag="i", doc="The output file from HAPCUT/HAPCUT2.") val input: FilePath,
   @arg(flag="o", doc="The output VCF with both phased and unphased variants.") val output: PathToVcf,
-  @arg(flag="r", doc="Output phased variants in GATK's ReadBackedPhasing format.") val gatkPhasingFormat: Boolean = false
+  @arg(flag="r", doc="Output phased variants in GATK's ReadBackedPhasing format.") val gatkPhasingFormat: Boolean = false,
+  @arg(flag="2", doc="The input is in HAPCUT2 format, otherwise HAPCUT") val hapCut2: Boolean = false
```

Comment:
Yeah, a larger refactor (probably not worth it just for auto-detection) would be to have the hapcut readers construct from Iterator[String] of lines .. then you could open the stream, wrap it in a buffered iterator, peek at it, and then pass it on down for consumption, without breaking stream-ability.  But that would be a bigger change for sure.